### DISCUSSION TITLE
Mania editor proposal: place notes with DFJK

### DISCUSSION THREAD

**Discussion by pfgithub** [ppy/osu#28955] [Ideas]
To place a chord right now, you have to click every individual spot you want a note at. It would be nice to be able to place notes in the mania editor just by pressing the keys for those notes. Also, placing holds could be supported by holding down a key, scrolling to the end point, and releasing.

Currently, J and K are used for "nudge selection", those keys would have to be changed if this is implemented so they don't overlap with the default mania keys. I understand these keys have been around for a long time so changing the defaults might be hard.

More mania editor things:

- Right now in the mania editor, there is one button for note and a second button for hold. There only needs to be one - dragging a note can create a hold. This would eliminate the need to be constantly switching between two tools when you want to place holds and regular notes together.
- In the editor, it would be very nice to hold a key to play and then release the key to return to where you started playing - this way you can test out a section and immediately be right back at the point you were editing to make any changes you need. I implemented this years ago here and it was really nice to edit with, but never tried to merge it: https://github.com/pfgithub/osu/commit/44f8d6ff1479bf762689c5d8ae1af4c0853bf7b0
- Often if you're scrolled to the position where some notes are, they will be invisible because the game thinks they are under the line. This makes it annoying to place notes because you can't see them get placed. It would be nice to have a little bit of buffer space around the line before they disappear or even to make notes not disappear in the editor when they're below the line
- Currently you can place two notes in one spot. Is there any reason for that? You don't want that in a completed map, so it should probably just do nothing or toggle if you try to place overlapping notes

**Comment by peppy** [comment]
I'm guessing what you really want is live-edit mode, where you can just play and get a beatmap. Definitely something we would see coming to the editor.

**Reply by Joehuu** [reply]
Already tracked in https://github.com/ppy/osu/issues/6608.

**Reply by peppy** [reply]
It's slightly different to the proposal so I'll unanswer for now just to keep things clear.

**Reply by pfgithub** [reply]
I think that issue is pretty much the same thing as long as the live keybinds are also available when editing the map normally
